tests/api.c

```diff
@@ -5645,6 +5649,306 @@ static void test_wolfSSL_dtls_export(void)
+    0x7E, 0xD3, 0xD0, 0xB5, 0x87, 0x1D, 0x81, 0x54,
+    0x14, 0x63, 0x00, 0x06, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x03
+};
+
+
+static byte canned_server_session[] = {
+    0xA7, 0xA4, 0x01, 0x40, 0x00, 0x41, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x80, 0x02, 0x00, 0x00, 0x00,
+    0x00, 0x80, 0x00, 0x1C, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x01, 0x01, 0x01, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xC0,
+    0x27, 0x08, 0x0F, 0x10, 0x01, 0x01, 0x00, 0x11,
+    0x05, 0x00, 0x01, 0x01, 0x01, 0x03, 0x03, 0x00,
+    0xBF, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x02, 0x00, 0x00, 0x00, 0x40, 0x00, 0x00, 0x00,
+    0x0A, 0x01, 0x01, 0x00, 0x20, 0x69, 0x11, 0x6D,
+    0x97, 0x15, 0x6E, 0x52, 0x27, 0xD6, 0x1D, 0x1D,
+    0xF5, 0x0D, 0x59, 0xA5, 0xAC, 0x2E, 0x8C, 0x0E,
+    0xCB, 0x26, 0x1E, 0xE2, 0xCE, 0xBB, 0xCE, 0xE1,
+    0x7D, 0xD7, 0xEF, 0xA5, 0x44, 0x80, 0x2A, 0xDE,
+    0xBB, 0x75, 0xB0, 0x1D, 0x75, 0x17, 0x20, 0x4C,
+    0x08, 0x05, 0x1B, 0xBA, 0x60, 0x1F, 0x6C, 0x91,
+    0x8C, 0xAA, 0xBB, 0xE5, 0xA3, 0x0B, 0x12, 0x3E,
+    0xC0, 0x35, 0x43, 0x1D, 0xE2, 0x10, 0xE2, 0x02,
+    0x92, 0x4B, 0x8F, 0x05, 0xA9, 0x4B, 0xCC, 0x90,
+    0xC3, 0x0E, 0xC2, 0x0F, 0xE9, 0x33, 0x85, 0x9B,
+    0x3C, 0x19, 0x21, 0xD5, 0x62, 0xE5, 0xE1, 0x17,
+    0x8F, 0x8C, 0x19, 0x52, 0xD8, 0x59, 0x10, 0x2D,
+    0x20, 0x6F, 0xBA, 0xC1, 0x1C, 0xD1, 0x82, 0xC7,
+    0x32, 0x1B, 0xBB, 0xCC, 0x30, 0x03, 0xD7, 0x3A,
+    0xC8, 0x18, 0xED, 0x58, 0xC8, 0x11, 0xFE, 0x71,
+    0x9C, 0x71, 0xD8, 0x6B, 0xE0, 0x25, 0x64, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0C,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x10, 0x00, 0x10, 0x00, 0x10, 0x00, 0x10,
+    0x00, 0x00, 0x06, 0x01, 0x04, 0x08, 0x01, 0x20,
+    0x28, 0x00, 0xC5, 0x8F, 0xA1, 0x59, 0x93, 0xEF,
+    0x7E, 0xD3, 0xD0, 0xB5, 0x87, 0x1D, 0x81, 0x54,
+    0x14, 0x63, 0x09, 0xE1, 0x50, 0x70, 0x02, 0x2F,
+    0x7E, 0xDA, 0xBD, 0x40, 0xC5, 0x58, 0x87, 0xCE,
+    0x43, 0xF3, 0x00, 0x06, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x04
+};
+
+
+static THREAD_RETURN WOLFSSL_THREAD tls_export_server(void* args)
+{
+    SOCKET_T sockfd = 0;
+    SOCKET_T clientfd = 0;
+    word16 port;
+
+    callback_functions* cbf;
+    WOLFSSL_CTX* ctx = 0;
+    WOLFSSL* ssl = 0;
+
+    char msg[] = "I hear you fa shizzle!";
+    char input[1024];
+    int idx;
+
+#ifdef WOLFSSL_TIRTOS
+    fdOpenSession(Task_self());
+#endif
+
+    ((func_args*)args)->return_code = TEST_FAIL;
+    cbf = ((func_args*)args)->callbacks;
+
+    {
+        WOLFSSL_METHOD* method = NULL;
+        if (cbf != NULL && cbf->method != NULL) {
+            method = cbf->method();
+        }
+        else {
+            method = wolfTLSv1_2_server_method();
```

Comment:
Hum, what if only TLS v1.3 is enabled? I see the cipher suite below is TLS v1.2 only... so you will need to add `WOLFSSL_NO_TLS12`.

```
./configure --enable-sessionexport --disable-tlsv12 && make
/usr/bin/ld: tests/unit_test-api.o: in function `test_wolfSSL_tls_export':
api.c:(.text+0x13678): undefined reference to `wolfTLSv1_2_server_method'
/usr/bin/ld: api.c:(.text+0x136d2): undefined reference to `wolfTLSv1_2_client_method'
/usr/bin/ld: tests/unit_test-api.o: in function `tls_export_server':
api.c:(.text+0x14679): undefined reference to `wolfTLSv1_2_server_method'
collect2: error: ld returned 1 exit status
```